src/components/proofs/EmailProof.tsx

```diff
@@ -67,7 +69,16 @@ export default function () {
     }
   }
 
-  async function onGenerateProof(secret?: string) {
+  async function onGenerateProof(secret: string) {
+    setError(undefined)
```

Comment:
Do we need to call this on every `onGenerateProof` call?

Reply:
Yes,  we need to reset that state. Also, moved it lower near `setLoading` state to reset it if  everything is okay